airgun/entities/computeresource.py

```diff
@@ -52,6 +52,7 @@ def delete(self, value):
     def list_vms(self, rhev_name, expected_vm_name=None):
         """Returns all the VMs on the CR or VM with specified name"""
         view = self.navigate_to(self, 'Detail', rhev_name=rhev_name)
+        view.virtual_machines.search(expected_vm_name)
```

Comment:
`expected_vm_name` is optional, so forcing this search will lead to searching for 'None' each time `expected_vm_name` is not specified. And btw `return` can be simplified as well:
```python
if expected_vm_name:
    view.virtual_machines.search(expected_vm_name)
    return view.virtual_machines.table.row(name=expected_vm_name)
return view.virtual_machines.table.rows()
```